src/providers/wfs/qgswfsprovider.cpp

```diff
@@ -1312,6 +1312,11 @@ bool QgsWFSProvider::empty() const
 #endif
   return !getFeatures( request ).nextFeature( f );
 
+}
+
+void QgsWFSProvider::handlePostCloneOperations( QgsVectorDataProvider *source )
+{
+  mShared = qobject_cast<QgsWFSProvider *>( source )->mShared;
```

Comment:
from issue https://github.com/qgis/QGIS/issues/48465
since the provider filter is in the shared data, what is the best approach?
* always make a deep copy?
* refactor the filter out?
ping @rouault @m-kuhn

Reply:
The reasoning was: https://github.com/qgis/QGIS/pull/46742#issuecomment-1007804956
where we tried to share the cache, not sure if that's preserved with a copy?
Can we detach on filter change?

Reply:
On setSubsetString(), probably partially clone the existing mShared data. By partially, it means for example *not* cloning mCacheIdDb, mCacheIdDbname and other member variables, so that the new mShared instance has a separate cache.